lib/tapioca/dsl/compilers/active_record_associations.rb

```diff
@@ -206,6 +206,16 @@ def populate_single_assoc_getter_setter(klass, association_name, reflection)
             "reset_#{association_name}",
             return_type: "void",
           )
+          if reflection.is_a?(ActiveRecord::Reflection::BelongsToReflection)
```

Comment:
Is there a good way of checking the version of Rails being targeted? We could avoid generating these signatures for pre-Rails 7 if so.

Reply:
I'm not sure if it's recommended but it looks like you could do [this](https://stackoverflow.com/a/19012037/26406)

```suggestion
          if reflection.is_a?(ActiveRecord::Reflection::BelongsToReflection) && Rails::VERSION::MAJOR >= 7
```

Reply:
Ooh, thanks! I'll look around the repo, see if there are other such checks like this and, if so, I'll add this one.

Reply:
Hmm, no such uses:

![screenshot of VS Code search for Rails::VERSION](https://github.com/user-attachments/assets/e887f7be-1f74-41a0-a77e-4e521303b056)

I do see a #rails_version used in tests, so I can investigate that.

Reply:
That worked great, thank you! I'll keep this in draft till I see whether tests will pass on CI (they passed locally).